test/spec/provider/cloud-element-templates/Validator.spec.js

```diff
@@ -410,6 +410,80 @@ describe('provider/cloud-element-templates - Validator', function() {
     });
 
 
+    describe('elementType', function() {
+
+      it('should accept elementType', function() {
+
+        // given
+        const templates = new Validator();
+
+        const templateDescriptor = require('./fixtures/elementType');
+
+        // when
+        templates.addAll(templateDescriptor);
+
+        // then
+        expect(errors(templates)).to.be.empty;
+
+        expect(valid(templates)).to.have.length(templateDescriptor.length);
+      });
+
+
+      it('should accept elementType if applyTo only contains superClass', function() {
+
+        // given
+        const templates = new Validator();
+
+        const templateDescriptor = require('./fixtures/elementType-superType');
+
+        // when
+        templates.addAll(templateDescriptor);
+
+        // then
+        expect(errors(templates)).to.be.empty;
+
+        expect(valid(templates)).to.have.length(templateDescriptor.length);
+      });
+
+
+      it('should reject if appliesTo is missing specified element type ', function() {
+
+        // given
+        const templates = new Validator();
+
+        const templateDescriptor = require('./fixtures/elementType-missing-target');
+
+        // when
+        templates.addAll(templateDescriptor);
+
+        // then
+        expect(errors(templates)).to.contain('template(id: <example.com.elementType-missing-target>, name: <Element Type (missing target)>): template does not apply to requested element type <bpmn:UserTask>');
+
+        expect(valid(templates)).to.be.empty;
+      });
+
+
+      it('should reject invalid replace targets', function() {
+
+        // given
+        const templates = new Validator();
+
+        const templateDescriptor = require('./fixtures/elementType-invalid-morphs');
+
+        // when
+        templates.addAll(templateDescriptor);
+
+        // then
+        expect(errors(templates)).to.contain('template(id: <example.com.TaskToGateway>, name: <Element Type (Task -> Gateway)>): can not morph <bpmn:ServiceTask> into <bpmn:ExclusiveGateway>');
```

Comment:
What is the reason for this limitations?

Reply:
We want to prevent the changing to broken types, like `Task -> Process` or `Task -> SequenceFlow`. As the simplest approach, we decided to only allow `Task -> Task`, `Gateway -> Gateway` and `Event -> Event`

cf. https://github.com/camunda/element-templates-json-schema/pull/48#discussion_r834478450

Reply:
Well, one way to approach this would also be "allow users to do what they want" (and see how the editing experience blows up in the process) :).

But I see how we rather want to be safe. 

If we got that route though, why do we choose `bpmn:Task` and not `bpmn:Activity` as the base for all "task nodes".

Reply:
> why do we choose `bpmn:Task` and not `bpmn:Activity`

Good point, I see no reason for it. I'll change it :+1: